primitives/src/pcs/mod.rs

```diff
@@ -77,54 +91,69 @@ pub trait PolynomialCommitmentScheme {
     /// allows for passing in any pointer type, e.g.: `trim(srs: &Self::SRS,
     /// ..)` or `trim(srs: Box<Self::SRS>, ..)` or `trim(srs: Arc<Self::SRS>,
     /// ..)` etc.
+    #[allow(clippy::type_complexity)]
     fn trim(
         srs: impl Borrow<Self::SRS>,
         supported_degree: usize,
         supported_num_vars: Option<usize>,
-    ) -> Result<(Self::ProverParam, Self::VerifierParam), PCSError>;
+    ) -> Result<
+        (
```

Comment:
Minor: What should we do for transparent PCS that has no SRS? 
Maybe we can also put the PCS's parameters into SRS still. Though PublicParameter seems to be a better name.

Reply:
Yes, just a naming issue, but we can keep as it is. Not that important.